web/src/server_events.js

```diff
@@ -239,8 +239,21 @@ function get_events({dont_block = false} = {}) {
             } catch (error) {
                 blueslip.error("Failed to handle get_events error", undefined, error);
             }
-            const retry_sec = Math.min(90, Math.exp(get_events_failures / 2));
-            get_events_timeout = setTimeout(get_events, retry_sec * 1000);
+
+            // We need to respect the server's rate-limiting headers, but beyond
+            // that, we also want to avoid contributing to a thundering herd if
+            // the server is giving us 500s/502s.
+            const backoff_delay_secs = Math.min(90, Math.exp(get_events_failures / 2));
```

Comment:
`Math.exp` is pretty arbitrary.  Can we make this an explicit backoff series with jitter, to match the other one?

Reply:
Heh, this algorithm dates to the first week of Zulip development: 0788f2efe60deb7cfc62e064553c477d80e1596d. 

Yeah that's probably reasonable.